### PR TITLE
Default value parameters and variadic detection 

### PR DESCRIPTION
**Pull request by leohubert**
Related to #66

### CODE REVIEW

File: transformer/src/getSignatureParameters.ts [hunk 1]
```diff
@@ -35,7 +35,9 @@ export function getSignatureParameters(signature: ts.Signature, context: Context
 		parameters.push({
 			n: parameterSymbol.getName(),
 			t: type && getTypeCall(type, type.symbol, context, getCtorTypeReference(parameterSymbol)) || getUnknownTypeCall(context),
-			o: declaration.questionToken !== undefined || declaration.initializer !== undefined,
+			o: declaration.questionToken !== undefined || declaration.initializer !== undefined || declaration.dotDotDotToken !== undefined,
+			var: !!declaration?.dotDotDotToken,
+			dv: type?.isClass() ? null /*TODO: handle class, but how ? */ : declaration.initializer,
```
Comment (leohubert) [inline transformer/src/getSignatureParameters.ts:40]: @Hookyns, do you know how to handle `private foo: Foo = new Foo()` here?

I can't find a documentation link for this part of typescript...

Comment (avin-kavish) [reply transformer/src/getSignatureParameters.ts:40]: What do you mean "handle it"? detecting runtime values of parameters is out of scope I think.

**Comment by Hookyns** [comment]
@leohubert is there any progress?

**Comment by leohubert** [comment]
> @leohubert is there any progress?

It's working fine excepted for this use case.
```typescript
class Test { 
   constructor(foo: Foo = new Foo()) { }
}
```
I don't have time to work on it right now but when I find some, I will continue the investigation :)

**Pull request closed without merging**